src/neptune_lightgbm/impl/__init__.py

```diff
@@ -372,15 +375,17 @@ def create_booster_summary(
             ax.imshow(image.imread(s))
             ax.axis("off")
             trees_series.append(neptune.types.File.as_image(ax.figure))
-        results_dict["{}trees".format(visuals_path)] = neptune.types.FileSeries(trees_series)
+        results_dict[f"{visuals_path}trees"] = neptune.types.FileSeries(trees_series)
 
     if log_trees_as_dataframe:
         if isinstance(booster, lgb.Booster):
             df = booster.trees_to_dataframe()
-            html_df = neptune.types.File.as_html(df)
-            results_dict["trees_as_dataframe"] = html_df
-            if not df.empty and not html_df.content:
-                warnings.warn("'trees_as_dataframe' wasn't logged. Probably generated dataframe was to large.")
+            stream_buffer = BytesIO()
+            df.to_csv(stream_buffer, index=False)
+            if sys.getsizeof(stream_buffer) < 32 * 1024 * 1024:
```

Comment:
If we'll introduce a fix in `neptune-client` this will still fail. Let's `try ... except ValueError ` instead :wink:

Reply:
which part will throw an error?
`neptune.types.File.from_stream(stream_buffer, extension="csv")` would fail only if the stream is > 32MB right?

Reply:
Yeah, I mean don't introduce strict buffer size checking and just cover `neptune-client` exception (it's a misfortune that it's just a `ValueError`). Instead, let's try something like:
```python
>>> try:
...     results_dict["trees_as_dataframe"] = neptune.types.File.from_stream(stream_buffer, extension="csv")
... except ValueError:
...     print('Test')
...
Your stream is larger than 32MB. Neptune supports saving files from streams smaller than 32MB.
Test
```